library/datasort.c

```diff
@@ -576,6 +641,21 @@ static int datasort_split(struct datasort_cfg *dcfg)
 	for (n = 0; n < dcfg->bctl_cnt; ++n) {
 		assert(dcfg->bctl[n] != NULL);
 
+		// If base is already sorted we don't need split it into chunks and sort it.
+		// Instead of that we can just iterate it over in the merge phase.
+		if (datasort_base_is_sorted(dcfg->bctl[n])) {
+			EBLOB_WARNX(dcfg->log, EBLOB_LOG_INFO, "defrag: datasort_add_view_chunk: start, index %d",
```

Comment:
I thought that it would be good to distinguish with next log write
https://github.com/interiorem/eblob/pull/36/files/e1619f2b2ba7acf63a2b100fa7c721bb0cab80b9#diff-1666230403901aa570972a85a4bb08f4R669

Reply:
It is helpful there because iteration could take some time